Unable to build documentation

@marktriggs Please see screen shot below - the building of the documentation is not working.

lyrasiss-mbp:archivesspace lyrasis$ build/run doc:build
Buildfile: /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/build/build.xml

set-classpath:

doc:yard:
     [java] Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=196m; support was removed in 8.0
     [java] Loading ArchivesSpace configuration file from path: uri:classloader:/config/config.rb
     [java] rake aborted!
     [java] NameError: uninitialized constant ArchivesSpaceOAIRepository::ArchivalObject
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/backend/app/lib/oai/aspace_oai_repository.rb:9:in `<class:ArchivesSpaceOAIRepository>'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/backend/app/lib/oai/aspace_oai_repository.rb:4:in `<main>'
     [java] uri:classloader:/jruby/kernel/kernel.rb:1:in `(root)'
     [java] uri:classloader:/jruby/kernel/kernel.rb:13:in `<main>'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/backend/app/lib/oai/aspace_oai_provider.rb:3:in `(root)'
     [java] uri:classloader:/jruby/kernel/kernel.rb:1:in `<main>'
     [java] uri:classloader:/jruby/kernel/kernel.rb:13:in `block in require_relative'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/backend/app/controllers/oai.rb:1:in `block in (root)'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/_yard/rakefile.rb:1:in `block in (root)'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/_yard/rakefile.rb:40:in `(root)'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/_yard/rakefile.rb:40:in `block in (root)'
     [java] /Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/_yard/rakefile.rb:74:in `block in (root)'
     [java] Tasks: TOP => doc:api
     [java] (See full trace by running task with --trace)

BUILD FAILED
/Users/lyrasis/Desktop/ASpaceVersions/check_new_master/archivesspace/build/build.xml:572: Java returned: 1

This will prevent us from being able to put out the final release but not the release candidates.

--- a/_yard/rakefile.rb
+++ b/_yard/rakefile.rb
@@ -37,7 +37,7 @@ namespace :doc do
 
     require_relative '../backend/app/lib/export'
 
-    Dir.glob(File.dirname(__FILE__) + '/../backend/app/controllers/*.rb') {|file| require file unless file =~ /system/}
+    Dir.glob(File.dirname(__FILE__) + '/../backend/app/controllers/*.rb').sort.each {|file| require file unless file =~ /system/}
 
     @endpoints = ArchivesSpaceService::Endpoint.all.sort{|a,b| a[:uri] <=> b[:uri]}
     @examples = JSON.parse( IO.read File.dirname(__FILE__) + "/../endpoint_examples.json" )

--- a/backend/app/controllers/container.rb
+++ b/backend/app/controllers/container.rb
@@ -1,4 +1,4 @@
-require_relative './search'
+require_relative 'search'
 
 class ArchivesSpaceService < Sinatra::Base
 

--- a/backend/app/controllers/location_profile.rb
+++ b/backend/app/controllers/location_profile.rb
@@ -1,3 +1,5 @@
+require_relative 'search'
+
 class ArchivesSpaceService < Sinatra::Base
 
   Endpoint.post('/location_profiles/:id')

--- a/backend/app/controllers/oai.rb
+++ b/backend/app/controllers/oai.rb
@@ -7,7 +7,7 @@ class ArchivesSpaceService < Sinatra::Base
     .params(["verb", String, "The OAI verb (Identify, ListRecords, GetRecord, etc.)"],
             ["metadataPrefix",
              String,
-             "One of: " + ArchivesSpaceOAIRepository::AVAILABLE_RECORD_TYPES.keys.join(", "),
+             "One of the supported metadata types.  See verb=ListMetadataFormats for a list.",
              :optional => true],
             ["from", String, "Start date (yyyy-mm-dd, yyyy-mm-ddThh:mm:ssZ)", :optional => true],
             ["until", String, "End date (yyyy-mm-dd, yyyy-mm-ddThh:mm:ssZ)", :optional => true],

--- a/backend/app/lib/oai/aspace_oai_repository.rb
+++ b/backend/app/lib/oai/aspace_oai_repository.rb
@@ -6,21 +6,25 @@ class ArchivesSpaceOAIRepository < OAI::Provider::Model
 
   FormatOptions = Struct.new(:record_types, :page_size)
 
-  AVAILABLE_RECORD_TYPES = {
-    'oai_dc' => FormatOptions.new([ArchivalObject], 25),
-    'oai_dcterms' => FormatOptions.new([ArchivalObject], 25),
-    'oai_marc' => FormatOptions.new([ArchivalObject], 25),
-    'oai_mods' => FormatOptions.new([ArchivalObject], 25),
-    'oai_ead' => FormatOptions.new([Resource], 1)
-  }
+  def self.available_record_types
+    {
+      'oai_dc' => FormatOptions.new([ArchivalObject], 25),
+      'oai_dcterms' => FormatOptions.new([ArchivalObject], 25),
+      'oai_marc' => FormatOptions.new([ArchivalObject], 25),
+      'oai_mods' => FormatOptions.new([ArchivalObject], 25),
+      'oai_ead' => FormatOptions.new([Resource], 1)
+    }
+  end
 
   # If a given record type supports deletes, we'll need a way to look up its
   # tombstone records.  Since we only know the URIs of those tombstones, we're
   # pretty much stuck doing slow lookups.
-  DELETE_LOOKUPS = {
-    Resource => Tombstone.where { Sequel.like(:uri, '%/resources/%') },
-    ArchivalObject => Tombstone.where { Sequel.like(:uri, '%/archival_objects/%') },
-  }
+  def self.delete_lookups
+    {
+      Resource => Tombstone.where { Sequel.like(:uri, '%/resources/%') },
+      ArchivalObject => Tombstone.where { Sequel.like(:uri, '%/archival_objects/%') },
+    }
+  end
 
   DELETES_PER_PAGE = 100
 
@@ -70,7 +74,7 @@ class ArchivesSpaceOAIRepository < OAI::Provider::Model
 
     metadata_prefix = options.fetch(:metadata_prefix)
 
-    format_options = AVAILABLE_RECORD_TYPES.fetch(metadata_prefix)
+    format_options = ArchivesSpaceOAIRepository.available_record_types.fetch(metadata_prefix)
     parsed_ref = JSONModel.parse_reference(uri)
 
     raise OAI::IdException.new if parsed_ref.nil?
@@ -96,9 +100,9 @@ class ArchivesSpaceOAIRepository < OAI::Provider::Model
     end
 
     resumption_token = if options.has_key?(:resumption_token)
-                         ArchivesSpaceResumptionToken.parse(options.fetch(:resumption_token), AVAILABLE_RECORD_TYPES)
+                         ArchivesSpaceResumptionToken.parse(options.fetch(:resumption_token), ArchivesSpaceOAIRepository.available_record_types)
                        else
-                         ArchivesSpaceResumptionToken.new(options, AVAILABLE_RECORD_TYPES)
+                         ArchivesSpaceResumptionToken.new(options, ArchivesSpaceOAIRepository.available_record_types)
                        end
 
     if resumption_token.state == ArchivesSpaceResumptionToken::PRODUCING_RECORDS_STATE
@@ -140,7 +144,7 @@ class ArchivesSpaceOAIRepository < OAI::Provider::Model
   end
 
   def options_for_type(metadata_prefix)
-    AVAILABLE_RECORD_TYPES.fetch(metadata_prefix)
+    ArchivesSpaceOAIRepository.available_record_types.fetch(metadata_prefix)
   end
 
   def build_set_description(text)
@@ -220,7 +224,7 @@ class ArchivesSpaceOAIRepository < OAI::Provider::Model
     # Get a dataset that will pull back all tombstones for the record types that
     # this metadata type supports.
     matching_tombstones = format_options.record_types.map {|record_type|
-      tombstone_ds = DELETE_LOOKUPS[record_type]
+      tombstone_ds = ArchivesSpaceOAIRepository.delete_lookups[record_type]
       if tombstone_ds
         restrict_tombstones_to_published_repositories(tombstone_ds)
       end

